### PR TITLE
Register clients with a template instead of shell/bconsole

### PR DESCRIPTION
**Pull request by pbouchez**
- Register clients with a template instead of shell/bconsole
- Add missing handler notify to reload Bareos server when changing conf
- Specify which catalog to delete in Backupcatalog job, to avoid a warning (no change, the default MyCatalog was picked)

### CODE REVIEW

File: tasks/register_client.yml [hunk 1]
```diff
@@ -1,15 +1,11 @@
 ---
-- name: Register client {{ item.name }}
-  shell: >
-    echo
-    "configure add client "
```
Comment (btravouillon) [inline tasks/register_client.yml:5]: From a previous discussion, I thought the command `configure add client ...` was adding the client to the DB.

If my understanding is correct here, a new client will be added to the database once the bareos server will be reloaded, correct?

Comment (pbouchez) [reply tasks/register_client.yml:5]: I guess (I haven't been able to confirm reading it anywhere)
We're simply following https://docs.bareos.org/IntroductionAndTutorial/Tutorial.html#manual-configuration-of-client : 2 methods can be used : either the `configure add` command, or creating the conf files and reloading.

Comment (btravouillon) [reply tasks/register_client.yml:5]: Nice! I definitely prefer your new proposal.

**Review by btravouillon — APPROVED** [review]
Looks good to me. 👍

**Review by milaflq — APPROVED** [review]

**Pull request merged**